### PR TITLE
update server Contact and Message Controllers for GET and POST, seed database

### PR DESCRIPTION
**Pull request by onelovelyname**

### CODE REVIEW

**Comment by jyeg** [comment]
cool mel!

File: server/messages/messageController.js [hunk 1]
```diff
@@ -1,45 +1,71 @@
...
   updateMessage: function (request, response) {
-    // var sentMessagesArray = request.body.sent;
-    // for (var i = 0; i < sentMessagesArray.length; i++) {
-    //   var query = { id: sentMessagesArray[i] };
-    //   Message.findOneAndUpdate(query, { status: 'sent' }, function (data) {
-        
-    //   });
-    // }
+    //update message in database after receiving message status
+    for (var status in request.body) {
```
Comment (kevhuang) [inline server/messages/messageController.js:34]: @onelovelyname I'm not clear on lines 34 and 35. What's the format of the incoming request? Typically you would use `for...in` loops if you wanted to iterate over properties in an object, not an array.

File: server/messages/messageController.js [hunk 2]
```diff
@@ -1,45 +1,71 @@
...
-  //   var newMessage = new Message(messageObject);
-  //   newMessage.save(function (error) {
-  //     if (error) { console.log(error); }
-  //   });
+  // logic to add or update Message after server receives POST request
+    var messageObject = {
+      userId: request.body.userId,
+      contactId: request.body.contactId,
+      text: request.body.text,
+      date: new Date(),
```
Comment (kevhuang) [inline server/messages/messageController.js:17]: @onelovelyname we'll need to remember to remove the hardcoding of `new Date()` and replace it with the proper code for setting the date/time entered by the user

File: server/messages/messageController.js (hunk 1 above)
Comment (onelovelyname) [inline server/messages/messageController.js:34]: @kevhuang The incoming PUT request to api/messages/ should be in this format: 

``` javascript
{ sent: [messageId, messageId],  
  deleted: [messageId, messageId]
}
```

Line 34 iterates through all properties in this object (i.e., 'sent' and 'deleted') and Line 35 assigns the array of sent or deleted messages to a variable. 

@jyeg - let me know if this accurately reflects the format of the request, or if there are changes required.

File: server/messages/messageController.js (hunk 2 above)
Comment (onelovelyname) [inline server/messages/messageController.js:17]: Agreed.

File: server/messages/messageController.js (hunk 1 above)
Comment (kevhuang) [inline server/messages/messageController.js:34]: Sweet, thanks for clarifying @onelovelyname.

**Comment by kevhuang** [comment]
@onelovelyname this isn't important for the build, but I don't think the seed files are set up correctly. At least I wasn't able to get it to work right off the bat.

I eventually got the seeding to work by doing the following:
1. Rename the seed files to contacts.json and messages.json (so that they get mapped to the same tables/collections that would be set up by the server)
2. Update contacts.json to be an array of objects

That got the sample data into the database, but obviously, the data types for fields like `userID`, `contactId`, and `date` are not correct.

**Comment by onelovelyname** [comment]
@kevhuang Re: setup of the seed files 
1. To clarify, are you saying that during setup, Mongoose defaults to naming collections with the pluralized form of the model name? Would you be able to provide documentation where you've seen this? We never explicitly declare the collection names to be plural, so I'm unsure how this happens unless it's automatic.

**Comment by kevhuang** [comment]
@onelovelyname yeah, if defaults to pluralized names. http://mongoosejs.com/docs/api.html#index_Mongoose-model

**Comment by kevhuang** [comment]
@mwharris333 please let us know if these changes look good

**Comment by mwharris1313** [comment]
Looks good. Go for it.

**Pull request merged**